typify-impl/src/merge.rs

```diff
@@ -77,12 +81,30 @@ fn merge_schema(a: &Schema, b: &Schema, defs: &BTreeMap<RefKey, Schema>) -> Sche
     try_merge_schema(a, b, defs).unwrap_or(Schema::Bool(false))
 }
 
+#[derive(Error, Debug)]
+pub(crate) enum SchemaMergeError {
+    #[error("Cannot merge two trivially false schemas")]
+    MergeWithFalse,
+    #[error("Error when trying to merge the two schema objects {a:?} and {b:?}: {source}")]
+    ObjectSchemaMerge {
+        source: Box<ObjectSchemaMergeError>,
+        a: SchemaObject,
+        b: SchemaObject,
```

Comment:
it seems sort of heavy to clone these `SchemaObjects`.. will the caller not already have access to them?

Reply:
Yes, but how will you know which two `SchemaObjects` couldn't be merged if you don't have them in the error?

Reply:
These clones only happen when there's an error. How big are these structs typically?

Reply:
While we use a `Result::Err` to indicate that the merge was not viable, this isn't an unexpected or aberrant condition. It doesn't lead to a failure to generate code from a schema. It's an expected and typical condition.